### PR TITLE
mrbayes: new package

### PR DESCRIPTION
**Pull request by akthoma**
adding new packages mrbayes and dependency libbeagle

### CODE REVIEW

File: var/spack/repos/builtin/packages/libbeagle/package.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+class Libbeagle(AutotoolsPackage):
+    """Beagle performs genotype calling, genotype phasing, imputation of
+       ungenotyped markers, and identity-by-descent segment detection."""
+
+    homepage = "https://github.com/beagle-dev/beagle-lib"
+    url      = "https://github.com/beagle-dev/beagle-lib/archive/beagle_release_2_1_2.tar.gz"
+
+    version('2_1_2', '1107614e86f652f8ee45c1c92f2af3d4')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/libbeagle/package.py]: could you please do `2.1.2` and add a custom `url_for_version` instead? I would say majority of packages use dot-separated versions.

**Comment by akthoma** [comment]
I edited libbeagle for the `url_for_version` fix and reinstalled the two so that I could see if it worked. However, the install for mrbayes failed. After a bunch of purging and cleaning and install attempts, I came to the conclusion that the `autoreconf` phase fails on a fresh download, but running `spack install mrbayes` after the fail yields a successful installation. In other words, the install succeeds only after running `spack install mrbayes` twice. Does anyone have any ideas on how to deal with this? The output from spack-build.out from the first attempt on a fresh download is below: 
```
==> Configure script not found: trying to generate it
==> Warning: *********************************************************
==> Warning: * If the default procedure fails, consider implementing *
==> Warning: *        a custom AUTORECONF phase in the package       *
==> Warning: *********************************************************
==> '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/libtool-2.4.6-gnegdqv3qu4om7rhgz3ks5tkd4as6klz/bin/libtoolize'
libtoolize: putting auxiliary files in '.'.
libtoolize: linking file './ltmain.sh'
libtoolize: You should add the contents of the following files to 'aclocal.m4':
libtoolize:   '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/libtool-2.4.6-gnegdqv3qu4om7rhgz3ks5tkd4as6klz/share/aclocal/libtool.m4'
libtoolize:   '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/libtool-2.4.6-gnegdqv3qu4om7rhgz3ks5tkd4as6klz/share/aclocal/ltoptions.m4'
libtoolize:   '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/libtool-2.4.6-gnegdqv3qu4om7rhgz3ks5tkd4as6klz/share/aclocal/ltsugar.m4'
libtoolize:   '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/libtool-2.4.6-gnegdqv3qu4om7rhgz3ks5tkd4as6klz/share/aclocal/ltversion.m4'
libtoolize:   '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/libtool-2.4.6-gnegdqv3qu4om7rhgz3ks5tkd4as6klz/share/aclocal/lt~obsolete.m4'
libtoolize: Remember to add 'LT_INIT' to configure.in.
libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.in,
libtoolize: and rerunning libtoolize and aclocal.
libtoolize: Consider adding '-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
==> '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/automake-1.15-5ozh5tbpekmsaims5k2wbkoh7behcddx/bin/aclocal'
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
==> '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/autoconf-2.69-chyigljkm2cbytiw3kkujwbebvu4cmff/bin/autoreconf' '-ivf' '-I' '/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/pkg-config-0.29.2-qwzakozc657b22swrkouj6nauay5ugdi/share/aclocal'
autoreconf: Entering directory `.'
autoreconf: configure.in: not using Gettext
autoreconf: running: aclocal -I /opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/pkg-config-0.29.2-qwzakozc657b22swrkouj6nauay5ugdi/share/aclocal --force 
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
autoreconf: configure.in: tracing
autoreconf: configure.in: not using Libtool
autoreconf: running: /opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/autoconf-2.69-chyigljkm2cbytiw3kkujwbebvu4cmff/bin/autoconf --include=/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/pkg-config-0.29.2-qwzakozc657b22swrkouj6nauay5ugdi/share/aclocal --force
autoreconf: running: /opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/autoconf-2.69-chyigljkm2cbytiw3kkujwbebvu4cmff/bin/autoheader --include=/opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/pkg-config-0.29.2-qwzakozc657b22swrkouj6nauay5ugdi/share/aclocal --force
autoheader: warning: missing template: BEAGLE_ENABLED
autoheader: Use AC_DEFINE([BEAGLE_ENABLED], [], [Description])
autoheader: warning: missing template: COMPLETIONMATCHES
autoheader: warning: missing template: MPI_ENABLED
autoheader: warning: missing template: SSE_ENABLED
autoheader: warning: missing template: THREADS_ENABLED
autoheader: warning: missing template: UNIX_VERSION
autoheader: warning: missing template: _64BIT
autoreconf: /opt/rit/app/spack/linux-rhel7-x86_64/gcc-7.1.0/autoconf-2.69-chyigljkm2cbytiw3kkujwbebvu4cmff/bin/autoheader failed with exit status: 1
```

**Comment by scheibelp** [comment]
> In other words, the install succeeds only after running spack install mrbayes twice.

When I tried this it looked like the autoreconf call fails the first time Spack tries to install. The second time, spack sees a "configure" file and doesn't attempt to run autoreconf (although you see ```==> Executing phase : 'autoreconf'``` on the command line it actually doesn't run autoreconf. In other words, the call to autoreconf in the first spack run fails but apparently generates a configure file that is used to perform a successful installation on the second try.

I don't know why the autoreconf on the first run fails.

File: var/spack/repos/builtin/packages/mrbayes/package.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    depends_on('libbeagle', when='+beagle')
+    depends_on('mpi', when='+mpi')
+
+    @property
+    def configure_directory(self):
+        return join_path(self.stage.source_path, 'src')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mrbayes/package.py]: You can replace this whole thing with:
```python 
configure_directory = 'src'
```

File: var/spack/repos/builtin/packages/mrbayes/package.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    def configure_args(self):
+        args = []
+        if '~beagle' in self.spec:
+            args.append('--with-beagle=no')
+        else:
+            args.append('--with-beagle=%s' % self.spec['libbeagle'].prefix)
+        if '~sse' in self.spec:
+            args.append('--enable-sse=no')
+        else:
+            args.append('--disable-sse')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mrbayes/package.py]: Aren't these both disabling?

File: var/spack/repos/builtin/packages/mrbayes/package.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        if '+mpi' in self.spec:
+            args.append('--enable-mpi=yes')
+        else:
+            args.appens('--disable-mpi')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mrbayes/package.py]: Typo: appens -> append

File: var/spack/repos/builtin/packages/libbeagle/package.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    version('2.1.2', '1107614e86f652f8ee45c1c92f2af3d4')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    depends_on('subversion')
+    depends_on('pkg-config')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libbeagle/package.py]: This is probably type build. Not sure about subversion.

**Pull request merged**